### PR TITLE
fix xmdf for hydro_as_2d

### LINKED ISSUE #313: [hydro-as-2d] driver support for the xmdf result files
Hi! 
We're using mdal/crayfish quite often, especially to show and analyse results of the 2d hydraulic model hydro-as-2d / hydro-ft-2d. ...and it would be great, if we can show and analyse the result from .h5 files in QGIS :) 

Is a driver for the xmdf (.h5) format planned to get into mdal?  maybe its just an extension to the already existing xmdf driver. hydro-as stores a relatively simple xmdf format with one group per dataset. Each group stores 4 tables (Mins, Maxs, Times, Values)
I'm not into cpp, but I wrote a simple function in python using the h5py lib (see attached) to read/write the data. May this could help a little bit to get the driver for reading these files into mdal :)

[hydras_h5_io.py.zip](https://github.com/lutraconsulting/MDAL/files/5458593/hydras_h5_io.py.zip)

test files
[testdata_hydro_as.zip](https://github.com/lutraconsulting/MDAL/files/5458647/testdata_hydro_as.zip)

it would be great to see the hydro-as xmdf support in the future!! 
cheers

### PR DESCRIPTION
**Pull request by vcloarec**
fix #313

### CODE REVIEW

File: mdal/frmts/mdal_xmdf.cpp [hunk 1]
```diff
@@ -117,6 +119,27 @@ bool MDAL::DriverXmdf::canReadDatasets( const std::string &uri )
...
+
+void MDAL::DriverXmdf::readGroupsTree( HdfFile &file, const std::string &name, MDAL::DatasetGroups &groups, size_t vertexCount, size_t faceCount ) const
+{
+  HdfGroup gMesh = file.group( name );
+  for ( const std::string &groupName : gMesh.groups() )
+  {
+    HdfGroup gGroup = gMesh.group( groupName );
+    if ( gGroup.isValid() )
+    {
+      if ( groupName == "Maximums" )
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.cpp:131]: whatabout Minimums?

File: mdal/frmts/mdal_xmdf.cpp [hunk 2]
```diff
@@ -193,35 +215,43 @@ void MDAL::DriverXmdf::addDatasetGroupsFromXmdfGroup( DatasetGroups &groups,
...
 std::shared_ptr<MDAL::DatasetGroup> MDAL::DriverXmdf::readXmdfGroupAsDatasetGroup(
-  const HdfGroup &rootGroup, const std::string &groupName, size_t vertexCount, size_t faceCount )
+  const HdfGroup &rootGroup, const std::string &groupName, size_t vertexCount, size_t faceCount ) const
 {
   std::shared_ptr<DatasetGroup> group;
   std::vector<std::string> gDataNames = rootGroup.datasets();
   if ( !MDAL::contains( gDataNames, "Times" ) ||
        !MDAL::contains( gDataNames, "Values" ) ||
-       !MDAL::contains( gDataNames, "Active" ) ||
+//       !MDAL::contains( gDataNames, "Active" ) ||
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.cpp:224]: remove commented out code

File: mdal/frmts/mdal_xmdf.hpp [hunk 3]
```diff
@@ -29,7 +29,7 @@ namespace MDAL
    * are stored in single
    *   3D arrays (time, x, y) for vector datasets
    *   2D arrays (time, x) for scalar datasets
-   *   2D arrays (time, active) for active flags
+   *   2D arrays (time, active) for active flags (supported by default)
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.hpp:32]: add "optional"

File: mdal/frmts/mdal_xmdf.hpp [hunk 4]
```diff
@@ -29,7 +29,7 @@ namespace MDAL
    * are stored in single
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.hpp:29]: also add to docs the XMDF is used by TUFLOW  AND Hydro-as. also write what is the difference between two (tuflow has active flag and single root)

File: tests/test_xmdf.cpp [hunk 5]
```diff
@@ -298,6 +299,161 @@ TEST( MeshXmdfTest, withReferenceTime )
   MDAL_CloseMesh( m );
 }
 
+TEST( MeshXmdfTest, HydroAs2D )
+{
+  // XMDF created with various TUFLOW utilities
```
Comment (PeterPetrik) [inline tests/test_xmdf.cpp:304]: remove?|

File: mdal/frmts/mdal_xmdf.hpp [hunk 6]
```diff
@@ -29,7 +29,7 @@ namespace MDAL
...
    *   3D arrays (time, x, y) for vector datasets
    *   2D arrays (time, x) for scalar datasets
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.hpp:31]: please fix https://github.com/lutraconsulting/MDAL/blob/master/README.md the description of XMDF*: format, add that it also supports hydro-as

File: mdal/frmts/mdal_xmdf.hpp [hunk 7]
```diff
@@ -29,7 +29,7 @@ namespace MDAL
    * are stored in single
    *   3D arrays (time, x, y) for vector datasets
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_xmdf.hpp:30]: Please update public docs https://www.mdal.xyz/drivers/xmdf.html

**Review by PeterPetrik — COMMENTED** [review]
well done!

File: mdal/frmts/mdal_xmdf.hpp (hunk 4 above)
Comment (PeterPetrik) [reply mdal/frmts/mdal_xmdf.hpp:29]: you may even use some of this https://github.com/lutraconsulting/MDAL/issues/313#issuecomment-718993039

File: mdal/frmts/mdal_xmdf.cpp (hunk 1 above)
Comment (vcloarec) [reply mdal/frmts/mdal_xmdf.cpp:131]: this part come from the former implementation... I guess there is no "Minimums" in TUFLOW file (not in test files).

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**